sunpy/net/dataretriever/tests/test_lyra_ud.py

```diff
@@ -15,6 +21,97 @@
 LCClient = lyra.LYRAClient()
 
 
+def mock_query_object(start_date, end_date):
+    """
+    Creation of a QueryResponse object, and prefill some
+    downloaded data from lyra.LYRAClient().fetch(Time('20 ..)
+    """
+    # Create a mock QueryResponse object
+    map_ = {
+        'TimeRange': TimeRange(parse_time(start_date), parse_time(end_date)),
+        'Time_start': parse_time(start_date),
+        'Time_end':  parse_time(end_date),
+        'source': 'Proba2',
+        'instrument': 'lyra',
+        'physobs': 'irradiance',
+        'provider': 'esa'
+    }
+
+    resp = QueryResponse.create(map_,
+    LCClient._get_url_for_timerange(TimeRange(parse_time(start_date), parse_time(end_date))))
+    # Attach the client with the QueryResponse
+    resp.client = LCClient
+    return resp
+
+
+@pytest.mark.remote_data
+def test_fetch_working():
+    """
+    Tests if the mock fetch contains the correct data.
+    """
+
+    qr1 = LCClient.search(a.Time('2012/10/4', '2012/10/6'),
+                          a.Instrument('lyra'))
+
+    # Create a mock query object
+    mock_qr = mock_query_object('2012/10/4', '2012/10/6')
+
+    mock_qr = mock_qr[0]
+    qr = qr1[0]
+    # Assert the values
+    assert mock_qr.source == qr.source
+    assert mock_qr.instrument == qr.instrument
+    assert mock_qr.physobs == qr.physobs
+    assert mock_qr.provider == qr.provider
+    assert mock_qr.url == qr.url
+    assert mock_qr.time == qr.time
+
+    # Assert if the time range is same
+    assert qr1.time_range() == TimeRange('2012/10/4', '2012/10/6')
+
+    # Assert the fetch object, and whether it returns the correct set of files
+    res = LCClient.fetch(qr1)
+    download_list = res.wait(progress=False)
+    assert len(download_list) == len(qr1)
+
+
+def daterange(start_date, end_date):
+    """
+    Helper function to create an iterable of datetime
+    containing each days as its entries from start to end
+    date
+    """
+
+    start_date = datetime.strptime(start_date, '%Y/%m/%d')
+    end_date = datetime.strptime(end_date, '%Y/%m/%d')
+    range_of_date = int((end_date - start_date).days)
+
+    if range_of_date is not 0:
+        for n in range(range_of_date+1):
+            yield start_date + timedelta(n)
+    else:
+        yield start_date
+
+
+# Create mocker fixtures for using parametrized tests
+@pytest.fixture
+def create_mock_fetch(mocker, sdate, edate, instrument):
+    mocker.patch('sunpy.net.fido_factory.Fido.fetch',
+        side_effect=(UnifiedResponse(mock_query_object(sdate, edate))))
+
+
+@pytest.fixture
+def create_mock_search(mocker, sdate, edate, instrument):
+    mocker.patch('sunpy.net.dataretriever.sources.lyra.LYRAClient.search',
+        return_value=mock_query_object(sdate, edate))
+
+
+@pytest.fixture
+def create_mock_result(mocker, sdate, edate, instrument):
+    mocker.patch('sunpy.net.download.Results.wait',
+    return_value=[LCClient._get_url_for_date(parse_time(date)) for date in daterange(sdate, edate)])
```

Comment:
This could read `LCClient._get_url_for_date(parse_time(TimeRange(sdate, edate).get_dates()))`